index.bs

```diff
@@ -2366,10 +2508,22 @@ partial interface MLGraphBuilder {
 <div>
     **Arguments:**
         - *a*: an {{MLOperand}}. The first input tensor.
-        - *b*: an {{MLOperand}}. The second input tensor.
+        - *b*: an {{MLOperand}}. The second input tensor when specified.
 
-    **Returns:** an {{MLOperand}}. The output tensor that contains the result of
-    element-wise comparison of the two input tensors.
+    **Returns:** an {{MLOperand}}. The output tensor that contains the result of element-wise comparison of the two input tensors.
+</div>
+<div>
+    **Operation types:**
+        - *equal*: Compare if the values of the two input tensors are equal, element-wise.
+        - *greater*: Compare if the values of the first input tensor is greater, element-wise.
+        - *greaterOrEqual*: Compare if the values of the first input tensor is greater or equal, element-wise.
+        - *lesser*: Compare if the values of the first input tensor is lesser, element-wise.
+        - *lesserOrEqual*: Compare if the values of the first input tensor is lesser or equal, element-wise.
+        - *not*: Invert the values of the input tensor to values 0 or 1, element-wise. Specifically, when the input value is non-zero, invert it to a {{boolean}} value 0. Conversely, for a zero input value, invert it to a {{boolean}} value. 
+</div>
+
+<div class="note">
+Although operations *greaterOrEqual* and *lesserOrEqual* can each be implemented in terms of operations *not*, *lesser*, and *greater* in other words `greater-or-equal(a, b)` is `not(lesser(a, b))`, they are specifically defined for performance reason to avoid double comparisons in each element-wise test.
```

Comment:
Happy to adjust the text, but know that NaN peculiar loosely defined behavior isn't limited to just this.

Reply:
I can take or leave it. It would be the first place we mention NaN behavior, which we do not mention elsewhere, and I suppose those users who know they need this behavior will already know they need it. Feel free to resolve.

Reply:
Along those lines, the note itself is not really needed IMHO (though it doesn't harm - but then why mention this only here). Once we define a method, it becomes an opportunity for optimizations in implementations.